composer/utils/batch_helpers.py

```diff
@@ -0,0 +1,99 @@
+from typing import Any
+
+from composer.core.types import BatchKeys
+
+
+def batch_get(batch: Any, key: BatchKeys) -> Any:
+    """Indexes into the batch given the key.
+
+    Tries all the common combination of batch and key pairs:
+        * int for sequence
+        * slice for sequence
+        * string for dict
+        * string atribute for object or named_tuple
+        * sequence of ints for sequence
+        * sequence of slices for sequence
+        * sequence of keys for dict
+        * sequence of attribute names for object or named_tuple
+
+    Args:
+        batch (Any): An object that contains the input and label of the items in the batch.
+            Can be any abritrary type that user creates, but we assume some sort of
+            sequence (list, tuple, tensor, array), mapping (dictionary),
+            or attribute store (object with data members, namedtuple).
+        key (BatchKeys): A key to index into the batch. BatchKeys is of type Union[BatchKey Sequence[BatchKey]]
+            and BatchKey is of type Union[int, str, slice].
+
+    Returns:
+        The part of the batch specified by the key. This could be any type depending on
+            what the batch is composed of.
+
+    Raises:
+        RuntimeError if the batch can't be indexed by the key.
+    """
+    try:
+        return batch[key]  # int or slice for sequence or string for dict.
+    except:
+        pass
+    try:
+        return [batch[k] for k in key]  # Sequence of ints, slices, or strings.
+    except:
+        pass
+    try:
+        return getattr(batch, key)  # Attribute for object or named tuple.
+    except:
+        pass
+    try:
+        return [getattr(batch, k) for k in key]  # Sequence of attributes for object or named tuple.
+    except:
```

Comment:
Can you also handle the case where the `key` is a "getter' callable or a (getter, setter) tuple? I think this will be helpful when updating the algorithms to use this function, so each algorithm individually does not have to condition what the key is.

Reply:
Any chance I could do that in a new PR? This PR already has a lot of stuff in it. I feel like we should just get it in?

Reply:
(Sorry, didn't see your reply here when I made the comment again below!) Sure, if you want, but I also don't think it'd be too hard to do it here....e.g.

```python3
if callable(key):  # key is a getter
  return key(batch)
if isinstance(key, tuple) and len(key) == 2 and callable(key[0]):  # (getter, setter) tuple
  return key[0](batch)
```

Reply:
ok done here: 
418c669